### PR TITLE
State all known limitations of right-associative extension methods

### PR DESCRIPTION
**Pull request by nicolasstucki**
[skip test]
[skip test_windows_fast]
[skip test_sbt]
[skip community_build]
[skip mima]

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Note that all limitations can be fixed with #19203

File: docs/_docs/reference/contextual/right-associative-extension-methods.md [hunk 1]
```diff
@@ -61,4 +61,36 @@ For instance, the `+::` method above would become
...
+```
+
+Another limitation of this representation is that it is impossible to pass the
+type parameters of the `def` explicitly. For instance:
+
+```scala
+  extension (x: Int)
+    def *:[T](xs: List[T]): List[T] = ...
+
+  xs.*:[Int](1) // error when trying to set T explicitly
```
Comment (bishabosha) [inline docs/_docs/reference/contextual/right-associative-extension-methods.md:84]: this still works:
```scala
*:[Int](List(1))(1)
```

Comment (nicolasstucki) [reply docs/_docs/reference/contextual/right-associative-extension-methods.md:84]: Yes. I only stated the cases that do not.

Comment (bishabosha) [reply docs/_docs/reference/contextual/right-associative-extension-methods.md:84]: right, but you state that its "impossible" - but it is possible, just not how might be expected

File: docs/_docs/reference/contextual/right-associative-extension-methods.md [hunk 2]
```diff
@@ -61,4 +61,36 @@ For instance, the `+::` method above would become
...
+
+  y.*:(ys) // error when following the parameter definition order
+  ys.*:(y)
+
+  *:(y)(ys) // error when following the parameter definition order
+  *:(ys)(y)
+```
+
+Another limitation of this representation is that it is impossible to pass the
+type parameters of the `def` explicitly. For instance:
```
Comment (bishabosha) [inline docs/_docs/reference/contextual/right-associative-extension-methods.md:78]:
```suggestion
type parameters of the `def` explicitly, (unless called in prefix form). For instance:
```
not sure if this is the best rewording, but gets the point accross

**Review by bishabosha — APPROVED** [review]

**Pull request merged**